### PR TITLE
documentation for HTMLMapElement.areas

### PR DESCRIPTION
**Pull request by ShubhamOulkar**
### Description

adding new documentation for HTMLMapElement.areas

It is part of https://github.com/mdn/mdn/issues/520

### Resource
[areas property](https://html.spec.whatwg.org/multipage/image-maps.html#dom-map-areas)

### CODE REVIEW

File: files/en-us/web/api/htmlmapelement/areas/index.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+browser-compat: api.HTMLMapElement.areas
+---
+
+{{ApiRef("HTML DOM")}}
+
+The **`areas`** property of the {{domxref("HTMLMapElement")}} return {{domxref("HTMLCollection")}} object of {{HTMLElement("area")}} elements present in the {{HTMLElement("map")}} element. This property only apply filter for `area` elements. The `areas` is readonly property so we can not assign a new collection to `areas` property, we can only access or iterate over existing areas collections.
+
+## Value
+
+An array-like object ({{domxref("HTMLCollection")}}) of elements.
```
[the remark this answers is not in the document]
Comment (ShubhamOulkar) [reply files/en-us/web/api/htmlmapelement/areas/index.md]: it is useful for beginners

**Review by ShubhamOulkar — COMMENTED** [review]
minor correction in content

File: files/en-us/web/api/htmlmapelement/areas/index.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+title: "HTMLMapElement: areas property"
+short-title: areas
+slug: Web/API/HTMLMapElement/areas
+page-type: web-api-instance-property
+browser-compat: api.HTMLMapElement.areas
+---
+
+{{ApiRef("HTML DOM")}}
+
+The **`areas`** read-only property of the {{domxref("HTMLMapElement")}} return {{domxref("HTMLCollection")}} object of {{HTMLElement("area")}} elements present in the {{HTMLElement("map")}} element. This property only apply filter for `area` elements. We can not assign a new collection to `areas` property, we can only access or iterate over existing areas collections.
```
Comment (teoli2003) [inline files/en-us/web/api/htmlmapelement/areas/index.md:11]: Read-only is clear enough: no need to explain you can't modify it.

```suggestion
The **`areas`** read-only property of the {{domxref("HTMLMapElement")}} elements returns a collection of {{HTMLElement("area")}} elements associated with the {{HTMLElement("map")}} element. 

Two successive calls to this property will return the same object.
```

File: files/en-us/web/api/htmlmapelement/areas/index.md [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+## Value
+
+An array-like {{domxref("HTMLCollection")}} object of `area` elements.
```
Comment (teoli2003) [inline files/en-us/web/api/htmlmapelement/areas/index.md:15]:
```suggestion
A live {{domxref("HTMLCollection")}} of {{domxref("HTMLAreaElement")}} objects.
```

File: files/en-us/web/api/htmlmapelement/areas/index.md [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+<img
+  usemap="#image-map"
+  src="left-right-arrow.png"
+  alt="left right arrow image" />
+```
+
+```js
+const mapElement = document.getElementsByName("image-map")[0];
+// Anchor elements are filtered out from HTMLCollection
+console.log(mapElement.areas); // output: HTMLCollection(2) [area, area]
```
Comment (teoli2003) [inline files/en-us/web/api/htmlmapelement/areas/index.md:45]:
```suggestion
console.log(mapElement.areas); // Output: HTMLCollection(2) [area, area]
```

File: files/en-us/web/api/htmlmapelement/areas/index.md [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+const mapElementAreas = mapElement.areas;
+// iterate over each area element and change target value to _blank
```
Comment (teoli2003) [inline files/en-us/web/api/htmlmapelement/areas/index.md:48]:
```suggestion
// Iterate over each area element and change target value to _blank
```

File: files/en-us/web/api/htmlmapelement/areas/index.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+[...mapElementAreas].forEach((area) => (area.target = "_blank"));
+console.log(mapElementAreas[0]);
+// click on left and right arrows, it will open a new tab.
```
Comment (teoli2003) [inline files/en-us/web/api/htmlmapelement/areas/index.md:51]: As a live example, I don't think to make them open in a new tab is good.

Can we find something better here?

**Review by teoli2003 — DISMISSED** [review]

File: files/en-us/web/api/htmlmapelement/areas/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (skyclouds2001) [reply files/en-us/web/api/htmlmapelement/areas/index.md]: But beside this, I don't think HTMLCollection is array-like

[the remark this answers is not in the document]
Comment (ShubhamOulkar) [reply files/en-us/web/api/htmlmapelement/areas/index.md]: This word is from mdn [HTMLCollection](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection). 
Htmlcollections are array like object similar to 'arguments'. 

I thought if it is helpful for readers to follow content from other sources on MDN. 

In my opinion, Generic collection or just HTMLCollection are the most correct words. What is your opinion?

File: files/en-us/web/api/htmlmapelement/areas/index.md (hunk 6 above)
Comment (ShubhamOulkar) [reply files/en-us/web/api/htmlmapelement/areas/index.md:51]: Then I will make dynamic changes in 'alt'.

File: files/en-us/web/api/htmlmapelement/areas/index.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (skyclouds2001) [reply files/en-us/web/api/htmlmapelement/areas/index.md]: Well, it seems ok for me now

[the remark this answers is not in the document]
Comment (ShubhamOulkar) [reply files/en-us/web/api/htmlmapelement/areas/index.md]: Ok, I will wait for @teoli2003 suggestions.

File: files/en-us/web/api/htmlmapelement/areas/index.md (hunk 3 above)
Comment (ShubhamOulkar) [reply files/en-us/web/api/htmlmapelement/areas/index.md:17]:
```suggestion
A  present {{domxref("HTMLCollection")}} of {{HTMLElement("area")}} objects.
```
`live` is not a proper word, what about `present' word?

**Comment by teoli2003** [comment]
Is this ready for re-review? Asking because the live sample is not working: https://pr31713.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/HTMLMapElement/areas

Anything I can do to help?

**Comment by ShubhamOulkar** [comment]
> Anything I can do to help?

Thank you for reporting @teoli2003.

Yes, help me out. Local server shows the image. I don't know what is happening with preview URL. 

image

**Comment by skyclouds2001** [comment]
Because the exact image is served on https://pr31713.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/HTMLMapElement/areas/left-right-arrow.png, but the demo is trying to get the image on https://live.mdnyalp.dev/en-US/docs/Web/API/HTMLMapElement/areas/left-right-arrow.png

**Comment by ShubhamOulkar** [comment]
Is it macro problem? Or this src url will get update after rendering. 
I don't know how these macro add <iframe > with src attribute.

**Review by Josh-Cena — APPROVED** [review]
I have updated the example with something that better represents the usage of `areas`. I have verified that the live example works locally. Since this has been open for so long, I won't delay it further. Thanks for the work!

**Pull request merged**